File: src/services/user-service/src/lib/redactor.go
package lib

func RedactPhoneNumber(phoneNumber string) string {
	runes := []rune(phoneNumber)
	if len(runes) <= 4 {
		return phoneNumber
	}
	redacted := ""
	for i := 0; i < len(runes)-4; i++ {
		if runes[i] == ' ' || runes[i] == '-' || runes[i] == '+' {
			redacted += string(runes[i])
		} else {
			redacted += "*"
		}
	}
	return redacted + string(runes[len(runes)-4:])
}

func RedactJWT(jwt string) string {
	runes := []rune(jwt)
	if len(runes) <= 10 {
		return "***"
	}
	return string(runes[:5]) + "..." + string(runes[len(runes)-5:])
}

File: src/services/user-bff/src/util/redactor.ts
export function redactPhoneNumber(phoneNumber: string): string {
  if (phoneNumber.length <= 4) {
    return phoneNumber;
  }
  let redacted = "";
  for (let i = 0; i < phoneNumber.length - 4; i++) {
    const char = phoneNumber[i];
    if (char === " " || char === "-" || char === "+") {
      redacted += char;
    } else {
      redacted += "*";
    }
  }
  return redacted + phoneNumber.slice(-4);
}

export function redactJWT(jwt?: string): string | undefined {
  if (!jwt) return undefined;

  if (jwt.length <= 10) {
    return "***";
  }
  return jwt.slice(0, 5) + "..." + jwt.slice(-5);
}
